homeassistant/components/blebox/__init__.py

```diff
@@ -0,0 +1,120 @@
+"""The BleBox devices integration."""
+import asyncio
+import logging
+
+from blebox_uniapi.error import Error
+from blebox_uniapi.products import Products
+from blebox_uniapi.session import ApiHost
+
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import CONF_HOST, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+from .const import DEFAULT_SETUP_TIMEOUT, DOMAIN, PRODUCT
+
+_LOGGER = logging.getLogger(__name__)
+
+PLATFORMS = ["cover"]
+
+PARALLEL_UPDATES = 0
+
+
+async def async_setup(hass: HomeAssistant, config: dict):
+    """Set up the BleBox devices component."""
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Set up BleBox devices from a config entry."""
+
+    websession = async_get_clientsession(hass)
+
+    host = entry.data[CONF_HOST]
+    port = entry.data[CONF_PORT]
+    timeout = DEFAULT_SETUP_TIMEOUT
+
+    api_host = ApiHost(host, port, timeout, websession, hass.loop, _LOGGER)
+
+    try:
+        product = await Products.async_from_host(api_host)
+    except Error as ex:
+        _LOGGER.error("Identify failed at %s:%d (%s)", api_host.host, api_host.port, ex)
+        raise ConfigEntryNotReady from ex
+
+    domain = hass.data.setdefault(DOMAIN, {})
+    domain_entry = domain.setdefault(entry.entry_id, {})
+    product = domain_entry.setdefault(PRODUCT, product)
+
+    for component in PLATFORMS:
+        hass.async_create_task(
+            hass.config_entries.async_forward_entry_setup(entry, component)
+        )
+
+    return True
+
+
+async def async_unload_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Unload a config entry."""
+    unload_ok = all(
+        await asyncio.gather(
+            *[
+                hass.config_entries.async_forward_entry_unload(entry, platform)
+                for platform in PLATFORMS
+            ]
+        )
+    )
+
+    if unload_ok:
+        hass.data[DOMAIN].pop(entry.entry_id)
+
+    return unload_ok
+
+
+async def async_create_blebox_entities(product, async_add, entity_klass, entity_type):
+    """Create entities from a BleBox product's features."""
+
+    entities = []
+    for feature in product.features[entity_type]:
+        entities.append(entity_klass(feature))
+
+    async_add(entities, True)
+    return True
+
+
+class BleBoxEntity:
```

Comment:
This should inherit `Entity`.
```suggestion
class BleBoxEntity(Entity):
```

Reply:
Already inherited indirectly later (via a different class), but added it for clarity anyway.